abusehelper/core/utils.py

```diff
@@ -2,23 +2,54 @@
 
 import csv
 import ssl
-import time
 import gzip
+import time
+import signal
 import socket
-import urllib2
 import httplib
+import numbers
+import urllib2
 import traceback
 import collections
 import email.parser
-
 import cPickle as pickle
 
 import idiokit
+
 from idiokit import heap
 from cStringIO import StringIO
+
 from . import events
 
 
+def _is_signal(name):
+    if not name.startswith("SIG") or name.startswith("SIG_"):
+        return False
+
+    signum = getattr(signal, name)
+
+    if not isinstance(signum, numbers.Integral):
+        return False
+
+    return True
+
+
+def signal_number_to_symbol(signum):
+
+    if not isinstance(signum, numbers.Integral):
+        return None
+
+    signames = filter(
+        lambda name: getattr(signal, name) == abs(signum),
+        filter(_is_signal, dir(signal))
+    )
+
+    if not signames:
+        return None
```

Comment:
We could just return ```signames``` even when it's empty, as the code's logic would still be consistent ("return names for the signum, if any").